### DISCUSSION TITLE
Can´t choose a good yuyv Resolution

### DISCUSSION THREAD

**Discussion by xSterney** [awawa-dev/HyperHDR#514] [Q&A]
Im close to finishing my Ambilight project using HyperHDR and HyperSPI.
It runs so smooth without any problems and im very happy about it. Now i wanted to fine tune the colours using the LUT Table. But there i ran into some difficulties...
I heard yuyv is superior to the mjpeg videformat, but for me yuyv doesnt work on my 
MS2109 [the one i bought](https://www.amazon.de/Newhope-Capture-Directly-Computer-Compatible-Schwarz/dp/B09NQM16VY/ref=mp_s_a_1_1_sspa?crid=218SSFVYDOWZO&keywords=newhope&qid=1676821829&sprefix=newhope%2Caps%2C141&sr=8-1-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9waG9uZV9zZWFyY2hfYXRm&psc=1) 
I select like a resolution of 1024x768 with a refreshrate of 60hz, but i only get like 640x480 with 10hz in the overview tab

![IMG_20230219_165443](https://user-images.githubusercontent.com/23609117/219959265-e96afbe5-7de6-4f45-aa2a-b8c3319869d8.jpg)

**Comment by awawa-dev** [comment]
You grabber MS2109 doesnt support such combination: 1024x768x60 YUV. Maybe did you mistake it with MS2130 https://github.com/awawa-dev/HyperHDR/discussions/499 ? Use the info panel which will show you supported combinations:
https://github.com/awawa-dev/HyperHDR/wiki/Quick-start#configuring-video--system-grabber
![obraz](https://user-images.githubusercontent.com/69086569/219959643-069b0144-f9a3-4c08-8202-09cfd83bf87b.png)

**Reply by xSterney** [reply]
Ahh thank you. Somehow i overlook the info button. So the highest refreshrate i can set on yuyv format is 30hz. 
On mjpeg i can choose 1024x768x60 with no problems. I dont have other formats like nv12...
Do you recommend me setting the LUT-Table up for the mjpeg format?

**Reply by awawa-dev** [reply]
nv12 is supported only by Windows for some reason. You can use standard ms2109 LUT table or you can calibrate it yourself for MJPEG (there is a PR that improves the result for MJPEG calibration: https://github.com/awawa-dev/HyperHDR/pull/508 , not merged yet). Also remember to set proper brightness/contrast/saturation which MS2109 requires before doing anything.

**Reply by xSterney** [reply]
thank you for your answer.
My englisch combined with technical things isnt the best, so i have to figure out some things.
So far my Ambilight works very well, im happy with the HyperSPI solution . Previous version with WLED was a bit laggy and not as responsive which was also a bit distracting then. Do you recommend the upgrade to ms2130? Especially if it comes to HDR content?

**Reply by awawa-dev** [reply]
MS2130 is a new product and I'm still testing it... haven't finished it yet: still the quality seems to be a bit better and USB3.0 support reduces latency especially for uncompressed YUV format (aprox. from ~80-100ms to 50-60ms). Your current MS2109 doesn't support USB3.0 and it's USB2.0 grabber underneath: Chinese engineers just added blue "USB3.0" port to increase the price and sell ;-) For sure an upgrade if our device has only USB2.0 ports is unprofitable, with USB3.0 ports available you can think about it

**Reply by xSterney** [reply]
yea i know they added the blueish thing for fake usb3. 0 haha. I think I played too much around with a few settings yesterday because now the HDR to SDR just show a blackscreen. Is there a file i can delete to get those settings to defaul again? Im asking for a friend :D
I also just orderer the new ms2130 :)

**Reply by awawa-dev** [reply]
sure, on Rpi if you use `pi` user:
```
sudo systemctl stop hyperhdr@pi.service
delete .hyperhdr folder in your home directory e.g. /home/pi/.hyperhdr (may be hidden)
sudo systemctl start hyperhdr@pi.service
```

**Reply by xSterney** [reply]
Thank you for your project and help, now everything works fine again. [test video from few days ago](https://www.youtube.com/watch?v=zYNKY4KsSUI)
I will start calibrating my usb capture when the ms2130 arrives...
i feel a bit dumb because i couldnt figure out the difference between LUT Table and LUT calibration...
I did the LUT calibration thing and now my HDR works fine again.
Also I like the fact that my Ambilight setup runs Dolby Vision without any problems and without any expensive stuff like the HDFuryDiva. I hope it will work with the new MS2130 too.

**Reply by awawa-dev** [reply]
> [test video from few days ago](https://www.youtube.com/watch?v=zYNKY4KsSUI)

very nice! 👍 

>  difference between LUT Table and LUT calibration...

LUT table tab is an old manual tool before the automatic LUT calibration was introduced.

Could you provide on pastebin.com your current HyperHDR logs taken 2-3 minutes after the restart? I want to verify something.

**Comment by xSterney** [comment]
fine so i just need to do the LUT-calibration and nothing else by myself right?
There you go:
https://pastebin.com/cC0ee5cf
I saw that the ESP32 to LED refreshrate is set to max 83hz in your Tutorial
I tested it with 120hz and had no issues, should i set it lower again?

**Reply by awawa-dev** [reply]
> fine so i just need to do the LUT-calibration and nothing else by myself right?

There are already prebuilt LUT tables for both MS2109 & MS2130. You can use them too.

**Comment by awawa-dev** [comment]
There is a slight delay in between TV vs LEDs on your video. Probably your current USB2.0 grabber adds something, but some settings can be improved in the configuration.

> 2023-02-20T21:46:55.713Z [PERFORMANCE] [USB capturing: FPS = 60.05, decoding = 32ms, frames = 3604, invalid = 0], [INSTANCE0: FPS = 60.05, processed = 3603], [LED0: FPS = 125.00, send = 7500, processed = 7500, dropped = 0]

You should lower your current high resolution because you dont need it for the ambient lighting and processing alone takes over 32ms per frame. Maybe 640 x 480 @ 60 fps mjpeg. You can also use 'quarter of frame' mode in the grabber to reduce it further. Smoothing is set to default 150ms, you can try to lower it a bit.

> I tested it with 120hz and had no issues, should i set it lower again?

Because you have small setup your LED strip can render such frame rate. But it has impact on the system to process frames, a system timer usually lacks such precision to invoke rendering every 8ms... it shouldn't be set over 100Hz.

**Comment by xSterney** [comment]
Yes a saw some lag in the ambilight test video too but now with your tipps i have a decoding time of 7ms and it runs very smooth
I set the Smoothing to 100ms. 
Is this okay or would you suggest this lower or higher or is it just more like a personal thing?
I set the refreshrate of the LEDs to 100hz too as you suggested. Now the next thing i have to do is to look for an option to get HDR to SDR tonemapping activated by alexa, mobile widget or the remote control. Is HyperHDR connectable with a hue bridge too because with it i also can control some powerplugs in my room (sadly no hue lights in my setup)

**Reply by awawa-dev** [reply]
> Is this okay or would you suggest this lower or higher or is it just more like a personal thing?

Only individual perception matters. One more thing: if you use HyperHDR only for movies (not gaming) than almost all the content is 23-24Hz. If it is not up-sampled by the video source then the capture rate for the grabber should be set to max. 30fps.

>  Is HyperHDR connectable with a hue bridge too because with it i also can control some powerplugs in my room (sadly no hue lights in my setup)

No, only lamps are supported so you need e.g. Home Assistant. BTW there is also a plugin for HyperHDR https://github.com/mjoshd/hyperhdr-ha If you search you will find here example how to activate HDR tone mapping remotely (e.g. Denon amplifies can provide the stream info and HyperHDR JSON API can be used here to switch HDR/SDR automatically)

**Reply by xSterney** [reply]
I think i let it on 60fps because i watch a lot of youtube which is often in 1080p60 or 2160p60. Sadly i dont have any of those option, i think i have to stay with changing it with my mobile phone via chrome browser. 
Is there a difference between your lut table and my lut table for the ms2109. Because i just did it with the automatic lut calibration program or is there any further things to adjust?

**Reply by awawa-dev** [reply]
In that case you'd better stay using 60fps capturing. When ms2130 arrives, switch from mjpeg to yuv. It  should reduce processing latency to 1-2ms. 

There is a calibration report. But it is only available right after its completion, it is not saved. If you had it, then you could compare it with my results (they are included in the ms2109 package, ms2130 is available in its thread).

**Comment by xSterney** [comment]
so yuv 60fps capturing is possible with the new ms2130 right?
Because the old ms2109 can only do 30 fps.
Sadly i didnt save the calibration report, but im very happy with it righ now.
Will there be a DolbyVision to SDR tone mapping in future, because the HDR->SDR doesnt work as good as it does in HDR content ( I think the colours of DolbyVision are even darker). 
I also wanted to say that the [l2c levelshifter](https://www.amazon.de/Converter-Pegelwandler-BiDirektional-Raspberry-Mikrocontroller/dp/B07DKT9JXT/ref=asc_df_B07DKT9JXT/?tag=googshopde-21&linkCode=df0&hvadid=309088177091&hvpos=&hvnetw=g&hvrand=10276289664286290556&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9042530&hvtargid=pla-644447451540&psc=1&th=1&psc=1&tag=&ref=&adgrpid=65229878087&hvpone=&hvptwo=&hvadid=309088177091&hvpos=&hvnetw=g&hvrand=10276289664286290556&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9042530&hvtargid=pla-644447451540) are suffiecient for Ambilight ( i already had them), so you dont have to buy the sn74ahct125n if you already have the l2c level shifter at home. Did some testing without smoothing and they are fast af too.

**Comment by awawa-dev** [comment]
> so yuv 60fps capturing is possible with the new ms2130 right?
Because the old ms2109 can only do 30 fps.

Yes, it does. I've just updated ms2130 review in its thread: the latency went down from ~100 to 50-65ms (for yuv) depending on the video source parameters.

> Will there be a DolbyVision to SDR tone mapping in future, because the HDR->SDR doesnt work as good as it does in HDR content ( I think the colours of DolbyVision are even darker).

DV is closed, proprietary solution and there is no USB grabber that can capture it properly and there is no software fix for that. Even expensive HDFury won't solve it. There is one exception: Dolby Vision in LLDV compatibility mode. Most video sources can provide it if required and HDFury should work but as users report also cheap Ezcoo splitter with modified firmware should work and some other devices too. https://www.avsforum.com/threads/alternative-devices-for-enabling-low-latency-dolby-vision-lldv.3254266/

> I also wanted to say that the [l2c levelshifter](https://www.amazon.de/Converter-Pegelwandler-BiDirektional-Raspberry-Mikrocontroller/dp/B07DKT9JXT/ref=asc_df_B07DKT9JXT/?tag=googshopde-21&linkCode=df0&hvadid=309088177091&hvpos=&hvnetw=g&hvrand=10276289664286290556&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9042530&hvtargid=pla-644447451540&psc=1&th=1&psc=1&tag=&ref=&adgrpid=65229878087&hvpone=&hvptwo=&hvadid=309088177091&hvpos=&hvnetw=g&hvrand=10276289664286290556&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9042530&hvtargid=pla-644447451540) are suffiecient for Ambilight ( i already had them), so you dont have to buy the sn74ahct125n if you already have the l2c level shifter at home. Did some testing without smoothing and they are fast af too.

It's not about the final visual effect speed, but about the electrical signal handling and its stability:
![obraz](https://user-images.githubusercontent.com/69086569/220583029-9c5d8abb-78ff-4e7d-982d-8697381d0afc.png)

**Reply by xSterney** [reply]
received the ms2130 one hour ago, but i cant select yuyv...i only have mjpeg
![Screenshot_2023-02-25-15-36-55-70_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/23609117/221362908-feb634b2-3ae8-44ee-b1d2-fa3d29e8a75d.jpg)

also it says its usb2.0 but i connected it to usb 3.0 for sure (the ones in the middle right?)
![Screenshot_2023-02-25-15-41-49-65_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/23609117/221363000-5c37a01e-5124-4abb-ab0c-4dd97adfeb79.jpg)

**Reply by awawa-dev** [reply]
Do you have an 'Undervoltage detected' warning on main HyperHDR page or any other devices connected to USB ports? It can usually appear after a few minutes of running the Rpi. Also run `lsusb` and check `dmesg` logs.

**Reply by awawa-dev** [reply]
> (the ones in the middle right?)

Yes, USB ports in the middle on Rpi4. But did you connect it directly or did you use some kind of USB extender?

**Reply by xSterney** [reply]
no i connected it directly. I tested all the ports and it still shows only mjpeg

here is lsusb
![image](https://user-images.githubusercontent.com/23609117/221365079-023815e9-eb4f-4f67-b49e-7a613b2098ce.png)

and for what should i look out for in the dmesg logs?
Can i pastebin them or is it "critical" data?

**Reply by awawa-dev** [reply]
You can send them on my email. The address is on my Github Profile. What about the undervoltage detected? How do you power your Rpi?

**Reply by awawa-dev** [reply]
Something is wrong with your system (classic undervolatge problem or Rpi is damaged):
```
pi@raspberrypi:~ $ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 1a86:55d4 QinHeng Electronics USB Single Serial
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```
when you execute lsusb you should see USB3.0 hub. This is my log from Rpi4 with no devices attached to usb3.0 (esp32 is connected to usb2.0).

**Reply by xSterney** [reply]
sorry i forgot to reply on your other questions.
Im not using any usb extender
I have a powersupply which is extra for the rpi i think that should work fine
![image](https://user-images.githubusercontent.com/23609117/221365800-00542aff-f0e0-4884-a707-1fa82b62ecf0.png)

This is the one i bought [ms2130](https://de.aliexpress.com/item/1005004990708725.html?spm=a2g0o.order_detail.order_detail_item.3.3cf96368C5W7wd&gatewayAdapt=glo2deu) its the correct one right?
I will send you the log per email

**Reply by awawa-dev** [reply]
OK, I will check it later. But check my message above: we sent messages in the same time.

**Reply by awawa-dev** [reply]
You have Rpi4 right?

**Reply by xSterney** [reply]
okay now i know the issue
I always thought the rpi 3b+ has usb3.0 too, but it hasn´t ://
Sorry my bad
So now the only thing is to stay on mjpeg 640x480@60 fps right?
I now try to load your LUT-Table (for mjpeg) into my pi

**Reply by awawa-dev** [reply]
Yes, this resolution and frame-rate is fine. Use the LUT I prepared for MJPEG stream. BTW Rpi4 should be available again on the market in 2-3 months. At least that's what the manufacturer claimed.

**Reply by xSterney** [reply]
Ah okay nice then i might upgrade in future
I downloaded that file. I researched and i found that it needs to be put in the ./hyperhdr folder right?
Is there a easy way to do it without pulling the sd card out of the rpi? Like over ssh or sth?
Edit:
Did it with MobaXterm and it was pretty easy. After uploading i restarted my pi and it works pretty good. Thanks!

**Reply by awawa-dev** [reply]
Great! Winscp should also work but it needs to enable 'show hidden files/directories' in its option first. Otherwise `\home\pi\.hyperhdr` folder won't be accessible in Winscp.

**Reply by xSterney** [reply]
okay fine:)
I tested DolbyVision LLDV and it still works for me (with the ezcoo matrix simple pro) without any issues. Can't say why it works for me and for other it doesn't...it was just plug and play for me.
Did you see my test video down below?

**Reply by awawa-dev** [reply]
In [thread on https://www.avsforum.com](https://www.avsforum.com/threads/alternative-devices-for-enabling-low-latency-dolby-vision-lldv.3254266/) there are some suggestions that Ezcoo incorporated this lldv feature in the factory firmware for some models. Still max brightness may needed to be adjusted since lldv doesn't carry such information and ezcoo probably calibrated it just for 350 nits. I dont use Dolby Vision so I'm not an expert in this subject ;) Your video looks fine, saturation is OK (not sure about the dark gray/black) the luminance can be tuned up in the processing tab. But it will be applied to all video content, not just lldv.

**Comment by xSterney** [comment]
Ah nice that sounds promising..
no I have the normal video stream on my tv and also amblight using dolby vision (might be the LLDV version) with this [ezcoo splitter](https://www.amazon.de/dp/B087JCN28V), but i did nothing special in settings or etc...for me it was working out of the Box. 
The colours are just very dark like watching HDR content.
Here is the proof [video](https://www.youtube.com/watch?v=-H7Wp62Up5w) 
(its just few days ago where i was using wled and not hyperspi so its a bit laggy)
With other splitters like the Feintech and an other noname one it didnt work. 
With them i had to decide if i want to watch DolbyVision Content or Ambilight.

**Reply by xSterney** [reply]
Since last week my USBs from the rpi doesnt work anymore.
First i was wondering why the ambilight is not working anymore, then i saw the rpi doesnt receive a picture. 
Also the esp32 wasn´t working anymore due to the fact i powered it over the rpi USBs.
I tried to connect a mouse and keyboard, but it didnt work too.
Are they simply turned off or are they defect? 
Do you know a way to check over this?

**Reply by awawa-dev** [reply]
Execute lsusb and dmesg having some USB devices connected.

**Reply by xSterney** [reply]
> Execute lsusb and dmesg having some USB devices connected.

![image](https://user-images.githubusercontent.com/23609117/230183113-070cf9f2-7fba-4346-b7fa-8078044260e3.png)

ms2130 is missing. tested it with the old ms2109 and it doesnt work too

**Reply by awawa-dev** [reply]
try to attach the USB keyboard and check again. what is in the dmesg?

**Reply by awawa-dev** [reply]
How do you power your Rpi?

**Reply by xSterney** [reply]
> try to attach the USB keyboard and check again. what is in the dmesg?

dmesg is very long....how can i send it to you? ( Getting a lot of "over-current change issues)
https://raspberrypi.stackexchange.com/questions/43609/what-could-be-causing-over-current-change-all-the-sudden
tried this but it doesnt work :/

https://www.amazon.de/Raspberry-Offizielles-Pi-Netzteil-schwarz/dp/B01DP8O5A4/ref=sr_1_5?keywords=raspberry+pi+power+supply&qid=1680723256&sr=8-5

I have this official powersupply

With USB Keyboard plugged in its no difference and the keyboard doesnt even light up

**Reply by awawa-dev** [reply]
> dmesg is very long....how can i send it to you? ( Getting a lot of "over-current change issues)

maybe upload it on pastebin.com

> tried this but it doesnt work :/

`max_usb_current` currently does nothing. It's already enabled for Raspberry Pi OS now.

> I have this official powersupply

Official power adapter or not, it doesn't mean it's not broken. Disconnect all devices except the USB keyboard from the Rpi and try another 2A USB power supply.

**Reply by xSterney** [reply]
> How do you power your Rpi?

i now tested it with a different rpi and everything works fine now...
Hope that stays like this. But why is the other rpi damaged now?

**Reply by awawa-dev** [reply]
If the USB keyboard fails, try starting the Rpi with no USB devices connected to see if the overcurrent error goes away then. Then attach e.g. esp32 and check dmesg again.

**Reply by awawa-dev** [reply]
> > How do you power your Rpi?
> 
> i now tested it with a different rpi and everything works fine now... Hope that stays like this. But why is the other rpi damaged now?

You must test them in the same conditions, with the same devices connected etc

**Reply by xSterney** [reply]
> pastebin.com

https://pastebin.com/6RMbz7rY
**no** device connected (no keyboard or ms2130) + only esp32 connected over gpios (hyperspi) but its **not** powered (i usually powered it over the usb ports of the rpi)

**Reply by awawa-dev** [reply]
disconnect all devices from rpi, the gpio as well. you have a short circuit somewhere, you are trying to feed esp32 via gpio or the power regulator is down.

**Reply by xSterney** [reply]
> disconnect all devices from rpi, the gpio as well. you have a short circuit somewhere, you are trying to feed esp32 via gpio or the power regulator is down.

disconnected all devices incl. the esp, but i still get those overcurrent errors.
Should i test if the other rpi works if i connect it to the gpios?

**Reply by awawa-dev** [reply]
If the Rpi is throwing the same error all the time with no devices connected and the same SD card & the power supply work fine for the other Rpi then it's probably broken.

**Reply by awawa-dev** [reply]
> usb 1-1-port2: over-current change

Check all USB ports, clean them, maybe something got into your port and short circuited. Only the second port generates these errors.

**Reply by xSterney** [reply]
> If the Rpi is throwing the same error all the time with no devices connected and the same SD card & the power supply work fine for the other Rpi then it's probably broken.

now ambilight works fine again with the other/fresh pi..

> > usb 1-1-port2: over-current change
> 
> Check all USB ports, clean them, maybe something got into your port and short circuited. Only the second port generates these errors.

will check it tomorrow. On first sight i don't see something wrong..

Thanks for your help🫶🏻
